packages/calcite-components/src/components/meter/meter.tsx

```diff
@@ -0,0 +1,522 @@
+  @Prop({ reflect: true }) low: number;
+
+  /** Specifies a high value.  When `fillType` is `"range"`, displays a different color when above the specified threshold.  */
+  @Prop({ reflect: true }) high: number;
+
+  @Watch("min")
+  @Watch("max")
+  @Watch("low")
+  @Watch("high")
+  @Watch("value")
+  handleRangeChange(): void {
+    this.calculateValues();
+    this.updateLabels();
+  }
+
+  /** Specifies the component's display, where `"single"` displays a single color and `"range"` displays a range of colors based on provided `low`, `high`, `min` or `max` values. */
+  @Prop({ reflect: true }) fillType: "single" | "range" = "range";
+
+  /** Specifies the Unicode numeral system used by the component for localization. */
+  @Prop() numberingSystem: NumberingSystem;
+
+  /** When `true`, number values are displayed with a group separator corresponding to the language and country format. */
+  @Prop({ reflect: true }) groupSeparator = false;
+
+  /** Specifies the appearance style of the component. */
+  @Prop({ reflect: true }) appearance: Extract<"outline" | "outline-fill" | "solid", Appearance> =
+    "outline-fill";
+
+  /** When either `rangeLabels` is `true`, specifies the format of displayed labels. */
+  @Prop({ reflect: true }) rangeLabelType: MeterLabelType = "percent";
+
+  /** When either `valueLabel` is `true`, specifies the format of displayed label. */
+  @Prop({ reflect: true }) valueLabelType: MeterLabelType = "percent";
+
+  /** When `labelType` is `"units"` and either `valueLabel` or `rangeLabels` are `true`, displays beside the `value` and/or  `min` values. */
+  @Prop() unitLabel: "";
+
+  /** When `true`, displays the values of `high`, `low`, `min`, and `max`. */
+  @Prop({ reflect: true }) rangeLabels: false;
+
+  /** When `true`, displays the current value. */
+  @Prop({ reflect: true }) valueLabel: false;
+
+  @Watch("rangeLabelType")
+  @Watch("valueLabelType")
+  @Watch("unitLabel")
+  @Watch("rangeLabels")
+  @Watch("valueLabel")
+  handleLabelChange(): void {
+    this.updateLabels();
+  }
+
+  /**
+   * Made into a prop for testing purposes only
+   *
+   * @internal
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated by t9n module
+  @Prop({ mutable: true }) messages: MeterMessages;
+
+  /**
+   * Use this property to override individual strings used by the component.
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated by t9n module
+  @Prop({ mutable: true }) messageOverrides: Partial<MeterMessages>;
+
+  @Watch("messageOverrides")
+  onMessagesChange(): void {
+    /* wired up by t9n util */
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  async componentWillLoad(): Promise<void> {
+    await setUpMessages(this);
+    setUpLoadableComponent(this);
+    this.calculateValues();
```

Comment:
Can `value`, `low`, `high` be computed if not specified at this point? This would align w/ `<meter>`'s behavior (see https://codepen.io/jcfranco/pen/dywyPQb?editors=1010).

This might also need to happen when certain props are modified at runtime.

This could be tackled in a follow-up PR if needed.

Reply:
To clarify - set `value` to "0" if not provided by a user?

Do we want to set an internal `low` and `high` even if not provided? 

Currently we don't have `optimum` as part of the API so the `low` and `high`, if unset by a user, are not used in any calculations, as they might be in native Meter.

Reply:
> To clarify - set value to "0" if not provided by a user?

It should match `min`, if not provided.

> Do we want to set an internal low and high even if not provided?

I was thinking setting the props directly. Do you see an issue in doing so?

> Currently we don't have optimum as part of the API so the low and high, if unset by a user, are not used in any calculations, as they might be in native Meter (we could add optimum in the future and address this then as well).

Sorry about the confusion, I added `optimum` in my codepen for exhaustiveness.

Reply:
Yeah, that makes sense. To clarify:

If no `low` provided, set `low` to `min`.
If no `high` provided, set `high` to `max`.
If no `value` provided, set `value` to `min`.

This can be set in `calculateValues()`, which is called in other component updates as well as `componentWillLoad()` - are there other cases we should check for?

Reply:
Updated to the above - added some checks for "out of bounds" values, and updated with some tests to check that various "improper" configurations are correctly adjusted and set.